core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/issue/ignore/scanner/IssueExclusionsRegexpScanner.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.analysis.issue.ignore.scanner;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.sonarlint.core.container.analysis.filesystem.FileMetadata.CharHandler;
+import org.sonarsource.sonarlint.core.container.analysis.filesystem.SonarLintInputFile;
+import org.sonarsource.sonarlint.core.container.analysis.issue.ignore.scanner.IssueExclusionsLoader.DoubleRegexpMatcher;
+
+public class IssueExclusionsRegexpScanner extends CharHandler {
+  private static final Logger LOG = Loggers.get(IssueExclusionsRegexpScanner.class);
+
+  private final StringBuilder sb = new StringBuilder();
+  private final List<Pattern> allFilePatterns;
+  private final List<DoubleRegexpMatcher> blockMatchers;
+  private final SonarLintInputFile inputFile;
+
+  private int lineIndex = 1;
+  private List<LineExclusion> lineExclusions = new ArrayList<>();
+  private LineExclusion currentLineExclusion = null;
+  private int fileLength = 0;
+  private DoubleRegexpMatcher currentMatcher;
+  private boolean ignoreAllIssues;
+
+  IssueExclusionsRegexpScanner(SonarLintInputFile inputFile, List<Pattern> allFilePatterns, List<DoubleRegexpMatcher> blockMatchers) {
+    this.allFilePatterns = allFilePatterns;
+    this.blockMatchers = blockMatchers;
+    this.inputFile = inputFile;
+    LOG.debug("Evaluate issue exclusions for '{}'", inputFile.relativePath());
+  }
+
+  @Override
+  public void handleIgnoreEoL(char c) {
+    if (ignoreAllIssues) {
```

Comment:
To avoid infinite loop, I introduced a local variable to short circuit. The loop was: IgnoreIssuesFilter -> InputFile.isIgnoreAllIssues -> checkMetadata -> initMetadata -> IssueExclusionsRegexpScanner -> InputFile.isIgnoreAllIssues